### PR TITLE
Download classroom roster

### PR DESCRIPTION
**Pull request by Kadaaran**
This PR is entended to fix this issue #1184 

A teacher can download a CSV file which contains a student's `identifier`, `github username`, `name`.

I'm not sure about putting all the logic into the Model, please don't hesitate if you have any ideas or suggestions ✨ 

screen shot 2018-01-26 at 21 50 29

### CODE REVIEW

File: app/models/roster_entry.rb [hunk 1]
```diff
@@ -6,4 +6,32 @@ class RosterEntry < ApplicationRecord
...
+        if entry.user
+          login = entry.user.github_user.login
+          name  = entry.user.github_user.name ? entry.user.github_user.name : ""
+        end
+
+        row << entry.identifier
+        row << login
+        row << name
+
+        roster_array << row
```
Comment (ashleytqy) [inline app/models/roster_entry.rb]: could you do without `row` and just have `roster_array << [entry.identifier, login, name]`?

File: app/models/roster_entry.rb [hunk 2]
```diff
@@ -6,4 +6,20 @@ class RosterEntry < ApplicationRecord
 
   validates :identifier, presence: true
   validates :roster,     presence: true
+
+  def self.to_csv
```
[the remark this answers is not in the document]
Comment (Kadaaran) [reply app/models/roster_entry.rb]: Don't know what to do with this one 🤔 I'll try something on Monday !

[the remark this answers is not in the document]
Comment (ashleytqy) [reply app/models/roster_entry.rb]: not sure if this helps the Assignment Branch Condition suggestion –
```
all.sort_by(&:identifier).each do |entry|
  github_user = entry.user.try(:github_user)
  login = github_user.try(:login) || ""
  name = github_user.try(:name) || ""
  roster_array << [entry.identifier, login, name]
end
```

[the remark this answers is not in the document]
Comment (Kadaaran) [reply app/models/roster_entry.rb]: I never heard about the `.try` method it works perfectly thank you @ashleytqy !
I'll definitely write it down 💪

File: app/controllers/orgs/rosters_controller.rb [hunk 3]
```diff
@@ -107,6 +109,14 @@ def add_student
       redirect_to roster_path(current_organization)
     end
 
+    def download_roster
+      respond_to do |format|
+        format.csv { send_data @roster_entries.to_csv, filename: "classroom_roster.csv", disposition: "attachment" }
+      end
+
+      return
```
Comment (d12) [inline app/controllers/orgs/rosters_controller.rb]: You can yank this `return` statement, it doesn't do anything here

Comment (Kadaaran) [reply app/controllers/orgs/rosters_controller.rb]: I did it so we won't have roster.csv in the url but it's the same without it indeed, thanks !

**Pull request merged**